azurerm/internal/services/firewall/firewall_policy_resource.go

```diff
@@ -142,6 +142,132 @@ func resourceFirewallPolicy() *pluginsdk.Resource {
 				},
 			},
 
+			"intrusion_detection": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"mode": {
+							Type: pluginsdk.TypeString,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.FirewallPolicyIntrusionDetectionStateTypeOff),
+								string(network.FirewallPolicyIntrusionDetectionStateTypeAlert),
+								string(network.FirewallPolicyIntrusionDetectionStateTypeDeny),
+							}, false),
+							Optional: true,
+						},
+						"signature_overrides": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"state": {
+										Type: pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(network.FirewallPolicyIntrusionDetectionStateTypeOff),
+											string(network.FirewallPolicyIntrusionDetectionStateTypeAlert),
+											string(network.FirewallPolicyIntrusionDetectionStateTypeDeny),
+										}, false),
+										Optional: true,
+									},
+									"id": {
+										Type:     pluginsdk.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"bypass_traffic_settings": {
```

Comment:
i think setting is redundant, and to match genrealy provider tense
```suggestion
						"traffic_bypass": {
```